src/layouts/_DevSiteNavigationLayout.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import IconSearch from 'terra-icon/lib/icon/IconSearch';
+import IconTile from 'terra-icon/lib/icon/IconTile';
+import { PrimaryNavigationLayout, NavigationItem } from '@cerner/terra-application/lib/layouts';
+import { PageContainer } from '@cerner/terra-application/lib/page';
+
+import DevSitePage from '../pages/_DevSitePage';
+import NotFoundPage from '../pages/_NotFoundPage';
+import SettingsModal from '../modals/_SettingsModal';
+import SearchModal from '../modals/_SearchModal';
+import ApplicationSwitcherModal from '../modals/_ApplicataionSwitcherModal';
+
+import siteConfigPropType from '../site/siteConfigPropTypes';
+import DevSiteSecondaryNavigation from './_DevSiteSecondaryNavigationLayout';
+
+const propTypes = {
+  /**
+   * The site config for the application.
+   */
+  siteConfig: siteConfigPropType.isRequired,
+};
+
+const DevSiteNavigationLayout = ({ siteConfig }) => {
+  const location = useLocation();
+  const history = useHistory();
+  const [showSettingsModal, setShowSettingsModal] = React.useState(false);
```

Comment:
Initialize all 3 of these to false for consistency?

Reply:
well, i'm actually cheating with the second two, they aren't set to true or false, but have an object with a render function placed in them, so we render based on the presence of an object.

Reply:
Could we rename them to settingsModalConfig or something instead?

Reply:
Renamed to `extensionModal` and `utilityModal`